project/src/main/webapp/script.js

```diff
@@ -400,3 +407,29 @@ function registerUserByToken() {
   }
   fetch('/register?idToken=' + googleUser.getAuthResponse().id_token, {method: 'POST'});
 }
+
+/** Sends feedback that includes the place the user chose. */
+function sendUserChoiceAsFeedback() {
+  let indexOfPlaceUserChose = getCheckedValueByElementId('chosen-place-form',
+      'Please select the place that you chose.');
+  if (indexOfPlaceUserChose <= recommendedPlaces.length) {
+    updateUserFeedback(recommendedPlaces, recommendedPlaces[indexOfPlaceUserChose].placeId,
+        false /** user tried again */)
+  }
+}
+
+/** Updates user feedback in the database, by using the feedback servlet. */
+function updateUserFeedback(recommendedPlaces, chosenPlaceId, tryAgain) {
```

Comment:
Also I think the name of this function is a little confusing. It stores the feedback right? It doesn't update a feedback already existing. So maybe change to storeUserFeedback.
Just a suggestion, do whatever you think.

Reply:
Great suggestion, done